tests/test_unit/test_cdf_tk/test_utils/test_http_client.py

```diff
@@ -403,6 +404,69 @@ def test_timeout_error(self, http_client_one_retry: HTTPClient, rsps: responses.
             FailedRequestItem(id=1, error="RequestException after 1 attempts (read error): Simulated timeout error")
         ]
 
+    @pytest.mark.usefixtures("disable_gzip")
```

Comment:
Can you add a test case to cover (edge) case when max_failures_before_abort is 0?